samples: net: google_iot_mqtt: Change read publish to blocking

JeffreyUrban opened the pull request:
Avoid endless loop in `case MQTT_EVT_PUBLISH`

Fixes #40965

Signed-off-by: Jeffrey Urban <JeffreyUrban@users.noreply.github.com>

rlubos commented on samples/net/cloud/google_iot_mqtt/src/protocol.c:
```diff
@@ -166,7 +166,7 @@ void mqtt_evt_handler(struct mqtt_client *const client,
+			bytes_read = mqtt_read_publish_payload_blocking(
 				client, d,
 				len >= 32 ? 32 : len);
 			if (bytes_read < 0 && bytes_read != -EAGAIN) {
```
I think we don't need to check for `bytes_read != -EAGAIN` here either, just break on any error.

rlubos reviewed (commented):
Thanks, we don't put the PR number in the commit titles though, if you add the follwoing to the commit body it shall automatically close the issue on merge:
```
Fixes #40965
```
Also, please use the following prefix in the title: `samples: net: google_iot_mqtt: `.

carlescufi commented:
> Signed-off-by: Jeffrey Urban [JeffreyUrban@users.noreply.github.com](mailto:JeffreyUrban@users.noreply.github.com)

You have to use a real email address.

JeffreyUrban commented:
I made the suggested changes.

rlubos reviewed (commented):
The second commit message exceeds the char/line limit (see the compliance result), this needs fixing (or just squash the commits, IMO there's no need for a separate ones).

I'm not sure about the other failures, @carlescufi any clue what's wrong in there (twister/clang)?

carlescufi commented:
> I'm not sure about the other failures, @carlescufi any clue what's wrong in there (twister/clang)?

No, can you rebase your branch on top of the latest `main` @JeffreyUrban?

carlescufi commented:
@JeffreyUrban 

```
Commit 5baa5f905d:
3: UC4 Line exceeds max length (80>75): "Fail on any error for mqtt_read_publish_payload_blocking, not excluding -EAGAIN."
```

JeffreyUrban committed:
samples: net: google_iot_mqtt: Change read publish to blocking

Avoid endless loop in `case MQTT_EVT_PUBLISH`
Fail on any error for mqtt_read_publish_payload_blocking, inc. -EAGAIN.

Fixes zephyrproject-rtos#40965

Signed-off-by: Jeffrey Urban <jeffrey@jeffreyurban.com>

JeffreyUrban commented:
Thank you for being patient. I rebased and squashed and set a commit-msg gitlint hook. It appears to be still failing clang/twister, and I don't know why.

rlubos commented:
@JeffreyUrban The CI faliures observed in this PR are not related to your changes, but rather to faulty `sample.yaml` defined for the sample. I've submitted #41103 to address the problem and will ask you to rebase once more after it gets merged.
